lib/linguist/languages.yml

```diff
@@ -3306,6 +3306,36 @@ Lex:
   tm_scope: source.lex
   ace_mode: text
   language_id: 199
+CameLIGO:
+  type: programming
+  extensions:
+  - ".mligo"
+  tm_scope: source.mligo
+  ace_mode: ocaml
+  codemirror_mode: mllike
+  codemirror_mime_type: text/x-ocaml
+  group: LigoLANG
```

Comment:
The group key won't do anything right now as there is no language listed here called 'LigoLANG'.
Are all variants of Ligo co-equal? If so then I'm not sure what would be done here if grouping is wanted, otherwise the other two should be grouped under the first.

Reply:
So your suggestion is to choose one and make it the "standard". If we don't want to I guess we need to make some change some change in Linguist ot support this :thinking: .

What if we make make a LigoLANG, but we only make the extension of configuration file, like `.json`, would that work then?

Reply:
> So your suggestion is to choose one and make it the "standard".

Yes, that would be best. From the look of the grammars you're adding, PascalLIGO may be a good candidate as it's simply named `ligo` in the grammar and has a scope of `source.ligo`; neither of these indicate the dialect.

You can also add an alias for those that want to override or use the name in codeblocks.

Alternatively, pick the most common dialect that people tend to use when referring to "LigoLANG".

> What if we make a LigoLANG language, but we only give the extension of configuration files, like `.json`, would that work then?

This is cheating 😉 ... and whilst it would "work", I don't recommend this as this is likely to upset users as most would expect a JSON file to be identified as such. Adding `.json` to another will also significantly increase the chances of misclassification of non-Ligo related uses, especially if a precise and 100% accurate heuristic can't be written.

Reply:
> So your suggestion is to choose one and make it the "standard".

Yes, that would be best. From the look of the grammars you're adding, PascalLIGO may be a good candidate as it's simply named `ligo` in the grammar and has a scope of `source.ligo`; neither of these indicate the dialect.

You can also add an alias for those that want to override or use the name in codeblocks.

Alternatively, pick the most common dialect that people tend to use when referring to "LigoLANG".

> What if we make a LigoLANG language, but we only give the extension of configuration files, like `.json`, would that work then?

This is cheating 😉 ... and whilst it would "work", I don't recommend this as this is likely to upset users as most would expect a JSON file to be identified as such. Adding `.json` to another will also significantly increase the chances of misclassification of non-Ligo related uses, especially if a precise and 100% accurate heuristic can't be written.

Reply:
(Your comment was sent twice)

Reply:
Following your advice, I put Pascaligo as default. This may change in the future